File: src/cs/Samples/Samples/Samples/FileProperties/ReportProperties.cs
using ApiSamples.Samples.SolidEdge;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace ApiSamples.Samples.FileProperties
{
    /// <summary>
    /// Reports file properties of a file.
    /// </summary>
    class ReportProperties
    {
        static void RunSample(bool breakOnStart)
        {
            if (breakOnStart) System.Diagnostics.Debugger.Break();

            SolidEdgeFileProperties.PropertySets propertySets = null;
            SolidEdgeFileProperties.Properties properties = null;
            SolidEdgeFileProperties.Property property = null;

            // Build path to file.
            string fileName = Path.Combine(InstallDataHelper.GetTrainingFolderPath(), "Coffee Pot.par");

            FileInfo fileInfo = new FileInfo(fileName);

            Console.WriteLine("Processing '{0}'.", fileName);

            try
            {
                if (fileInfo.Exists == false)
                {
                    throw new FileNotFoundException("File not found.", fileName);
                }

                // Create a new instance of PropertySets.
                propertySets = new SolidEdgeFileProperties.PropertySets();

                // Open the file in readonly mode.
                propertySets.Open(fileName, true);

                // Loop through the PropertySets.
                for (int i = 0; i < propertySets.Count; i++)
                {
                    // Get a reference to the current PropertySet.
                    properties = (SolidEdgeFileProperties.Properties)propertySets[i];

                    Console.WriteLine(properties.Name);

                    // Loop through the properties for the current PropertySet.
                    for (int j = 0; j < properties.Count; j++)
                    {
                        // Get a reference to the current PropertySet.
                        property = (SolidEdgeFileProperties.Property)properties[j];

                        object propertyValue = null;
                        string typeName = "Unknown";

                        // property.Value can throw an exception!
                        try
                        {
                            // Attempt to get the property value.
                            propertyValue = property.Value;
                            typeName = propertyValue.GetType().FullName;
                        }
                        catch (System.Exception ex)
                        {
                            propertyValue = ex.Message;
                        }

                        Console.WriteLine("\t{0} ({1}) = '{2}'", property.Name, typeName, propertyValue);
                    }
                }
            }
            catch (System.Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            finally
            {
                if (propertySets != null)
                {
                    propertySets.Close();
                }
            }
        }
    }
}

File: src/vb/Samples/Samples/Samples/FileProperties/ReportProperties.vb
Imports ApiSamples.Samples.SolidEdge
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Text

Namespace ApiSamples.Samples.FileProperties
	''' <summary>
	''' Reports file properties of a file.
	''' </summary>
	Friend Class ReportProperties
		Private Shared Sub RunSample(ByVal breakOnStart As Boolean)
			If breakOnStart Then
				System.Diagnostics.Debugger.Break()
			End If

			Dim propertySets As SolidEdgeFileProperties.PropertySets = Nothing
			Dim properties As SolidEdgeFileProperties.Properties = Nothing
			Dim [property] As SolidEdgeFileProperties.Property = Nothing

			' Build path to file.
			Dim fileName As String = Path.Combine(InstallDataHelper.GetTrainingFolderPath(), "Coffee Pot.par")

			Dim fileInfo As New FileInfo(fileName)

			Console.WriteLine("Processing '{0}'.", fileName)

			Try
				If fileInfo.Exists = False Then
					Throw New FileNotFoundException("File not found.", fileName)
				End If

				' Create a new instance of PropertySets.
				propertySets = New SolidEdgeFileProperties.PropertySets()

				' Open the file in readonly mode.
				propertySets.Open(fileName, True)

				' Loop through the PropertySets.
				For i As Integer = 0 To propertySets.Count - 1
					' Get a reference to the current PropertySet.
					properties = DirectCast(propertySets(i), SolidEdgeFileProperties.Properties)

					Console.WriteLine(properties.Name)

					' Loop through the properties for the current PropertySet.
					For j As Integer = 0 To properties.Count - 1
						' Get a reference to the current PropertySet.
						[property] = DirectCast(properties(j), SolidEdgeFileProperties.Property)

						Dim propertyValue As Object = Nothing
						Dim typeName As String = "Unknown"

						' property.Value can throw an exception!
						Try
							' Attempt to get the property value.
							propertyValue = [property].Value
							typeName = propertyValue.GetType().FullName
						Catch ex As System.Exception
							propertyValue = ex.Message
						End Try

						Console.WriteLine(ControlChars.Tab & "{0} ({1}) = '{2}'", [property].Name, typeName, propertyValue)
					Next j
				Next i
			Catch ex As System.Exception
				Console.WriteLine(ex.Message)
			Finally
				If propertySets IsNot Nothing Then
					propertySets.Close()
				End If
			End Try
		End Sub
	End Class
End Namespace
